### PR TITLE
A bit "smarter" extended selection checkbox toggle. #1676

### PR DESCRIPTION
**Pull request by rozyczko**
- If the checkbox is clicked in a row which is NOT selected (dark background), only this row gets toggled
- If the checkbox is clicked in a row which is selected (one or many rows can be selected) all rows get toggled based on the content of the checkbox clicked.

This is hopefully less confusing for single row checks while maintaining extended selection applicability.

### CODE REVIEW

File: src/sas/qtgui/Perspectives/Fitting/FittingWidget.py [hunk 1]
```diff
@@ -1061,11 +1061,19 @@ def selectedParameters(self):
         return [s.row() for s in self.lstParams.selectionModel().selectedRows()
                 if self.isCheckable(s.row())]
 
-    def setParameterSelection(self, status=QtCore.Qt.Unchecked):
+    def setParameterSelection(self, status=QtCore.Qt.Unchecked, item=None):
         """
         Selected parameters are chosen for fitting
         """
         # Convert to proper indices and set requested enablement
+        if item is None:
```
Comment (krzywon) [inline src/sas/qtgui/Perspectives/Fitting/FittingWidget.py:1069]: This breaks the '(De-)Select parameters for fitting' options when right-clicking on parameters. The `selectParameters()` and `deselectParameters()` methods will need to pass `item` somehow.

**Review by krzywon — CHANGES_REQUESTED** [review]
The code looks fine and the functionality works as described, but this breaks something (see my comment).

File: src/sas/qtgui/Perspectives/Fitting/FittingWidget.py (hunk 1 above)
Comment (rozyczko) [reply src/sas/qtgui/Perspectives/Fitting/FittingWidget.py:1071]: Great catch! Fixed.

**Comment by RichardHeenan** [comment]
In my current local build of this branch, the mutual constrain does not work, is there a commit of fix above needed still?  I get error
11:59:36 - ERROR: Traceback (most recent call last):
  File "C:\sasview42\sasview\src\sas\qtgui\Perspectives\Fitting\FittingWidget.py", line 757, in showMultiConstraint
    self.addConstraintToRow(constraint=constraint, row=row)
  File "C:\sasview42\sasview\src\sas\qtgui\Perspectives\Fitting\FittingWidget.py", line 810, in addConstraintToRow
    symbol_dict = self.parent.perspective().getSymbolDictForConstraints()
  File "C:\sasview42\sasview\src\sas\qtgui\Perspectives\Fitting\FittingPerspective.py", line 537, in getSymbolDictForConstraints
    symbol_dict.update(tab.getSymbolDict())
  File "C:\sasview42\sasview\src\sas\qtgui\Perspectives\Fitting\FittingWidget.py", line 4387, in getSymbolDict
    model_name = self.kernel_module.name
AttributeError: 'NoneType' object has no attribute 'name'

**Comment by rozyczko** [comment]
I don't see this failure - what are you constraining, @RichardHeenan ?

**Review by krzywon — APPROVED** [review]

**Comment by krzywon** [comment]
The issues noted by @RichardHeenan might need to be addressed before merging, but this should otherwise be ready.

**Comment by rozyczko** [comment]
I am not able to see the issue @RichardHeenan is reporting and looking at the code it doesn't seem the current change could cause the above error.
I'll create another issue for the observed bug and close this PR.

**Pull request merged**